### PR TITLE
Add pattern constraint to `in_manifest` on `DataObject`

### PR DESCRIPTION
**Pull request by aclum**
This PR adds a pattern constraint to in_manifest. Without this it is possible to malform the id mention such that it causes voliations for refscan and for linkml_runtime/utils/metamodelcore.py in nmdc_automation.

Fixes https://github.com/microbiomedata/nmdc-schema/issues/2274

Related to https://github.com/microbiomedata/issues/issues/948

### CODE REVIEW

File: src/data/invalid/DataObject-in_manifest-invalid.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#this test should fail because the value of in_manifest should be a list, not a list of lists
```
Comment (eecavanna) [inline src/data/invalid/DataObject-in_manifest-invalid.yaml:1]: The JSON equivalent of the `in_manifest` value below is:

```json
  "in_manifest": [
    "['nmdc:manif-11-7796sg87']"
  ]
```

It is already a list of strings (as opposed to a list of lists)—but the string, itself, does not have the structure I think schema authors expect.

**Comment by aclum** [comment]
cc @mbthornton-lbl  FYI

File: src/schema/basic_classes.yaml [hunk 2]
```diff
@@ -305,7 +305,11 @@ classes:
         structured_pattern:
           syntax: "^{id_nmdc_prefix}:(wfmag|wfmb|wfmgan|wfmgas|wfmsa|wfmp|wfmt|wfmtan|wfmtas|wfnom|wfrbt|wfrqc)-{id_shoulder}-{id_blade}{id_version}$|^{id_nmdc_prefix}:(omprc|dgms|dgns)-{id_shoulder}-{id_blade}$"
           interpolated: true
-
+      in_manifest:
```
Comment (kheal) [inline src/schema/basic_classes.yaml:308]: Can we add the `structured_pattern` to the global definition instead?  That way the `range` and the `structured_pattern` are stated in the same spot.

**Review by kheal — APPROVED** [review]

**Review by bmeluch — APPROVED** [review]

File: src/schema/basic_classes.yaml [hunk 3]
```diff
@@ -305,7 +305,7 @@ classes:
         structured_pattern:
           syntax: "^{id_nmdc_prefix}:(wfmag|wfmb|wfmgan|wfmgas|wfmsa|wfmp|wfmt|wfmtan|wfmtas|wfnom|wfrbt|wfrqc)-{id_shoulder}-{id_blade}{id_version}$|^{id_nmdc_prefix}:(omprc|dgms|dgns)-{id_shoulder}-{id_blade}$"
           interpolated: true
-
+
```
Comment (eecavanna) [inline src/schema/basic_classes.yaml:308]: I think the addition of this space was unintentional. Committing this suggestion will remove it, simplifying the PR diff.

```suggestion

```

**Pull request merged**